environs/tools/tools.go

```diff
@@ -215,12 +229,22 @@ func convertToolsError(err *error) {
 	}
 }
 
-// PreferredStream returns the tools stream used to search for tools, based
-// on the required version, whether devel mode is required, and any user specified stream.
-func PreferredStream(vers *version.Number, forceDevel bool, stream string) string {
+var streamFallbacks = map[string][]string{
+	ReleasedStream: {ReleasedStream},
+	ProposedStream: {ProposedStream, ReleasedStream},
+	DevelStream:    {DevelStream, ProposedStream, ReleasedStream},
+	TestingStream:  {TestingStream, DevelStream, ProposedStream, ReleasedStream},
+}
+
+// PreferredStreams returns the tools streams that should be searched
+// for tools, based on the required version, whether devel mode is
+// required, and any user specified stream. The streams are in
+// fallback order - if there are no matching tools in one stream the
+// next should be checked.
+func PreferredStreams(vers *version.Number, forceDevel bool, stream string) []string {
 	// If the use has already nominated a specific stream, we'll use that.
 	if stream != "" && stream != ReleasedStream {
-		return stream
+		return copyStrings(streamFallbacks[stream])
```

Comment:
if the user specifies a stream not in streamFallbacks, we should just use that stream verbatum

Reply:
Done.